experiment/molecule/src/common.ts

```diff
@@ -14,55 +15,303 @@ export const Uint8: FixedBinaryCodec<number> = {
     return new DataView(buf).getUint8(0);
   },
 };
-export declare const HexUint8: FixedBinaryCodec<string>;
+export const HexUint8: FixedBinaryCodec<string> = {
+  __isFixedCodec__: true,
+  byteLength: 1,
+  pack(str) {
+    return Uint8.pack(parseInt(str, 16));
+  },
+  unpack(buf) {
+    return `0x${Uint8.unpack(buf).toString(16)}`;
+  },
+};
 
 // array Uint16 [byte; 2]
+export const Uint16LE: FixedBinaryCodec<number> = {
+  __isFixedCodec__: true,
+  byteLength: 2,
+  pack(u8) {
+    const buf = new ArrayBuffer(2);
+    new DataView(buf).setUint16(0, u8, true);
```

Comment:
we should check if the pack number is a valid u16 number at first, `U16_MIN <= u16 <= U16_MAX`

Reply:
done